Detections/SigninLogs/DiscoveryAttemptWithAzureHound.yaml

```diff
@@ -0,0 +1,53 @@
+id: 25a3fd0b-8fc1-42b9-8eca-867477218aea
+name: Azure AD Discovery attempt with AzureHound
+description: |
+  'The AzureHound collector for BloodHound (https://bloodhound.readthedocs.io/en/latest/data-analysis/bloodhound-gui.html) requires authentication using Connect-AzureAD and Connect-AZAccount. This detection checks for successfull authentication to Azure Active Directory PowerShell and then within 5 minutes also to Microsoft Azure PowerShell to try and identify possible use of AzureHound.'
+severity: Low
+requiredDataConnectors:
+  - connectorId: SignInLogs
+    dataTypes:
+      - SignInLogs
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Discovery
+relevantTechniques:
+  - T1A0007
+query: |
+  let TimeDelta = 5m; //Maximum amount of time between an actor running Connect-AzureAD and Connect-AzAccount  
+  let ConnectAzureADEvents = SigninLogs 
+    | where AppDisplayName == "Azure Active Directory PowerShell" and ResultType == "0" and IsInteractive == true
+    | extend AADSignInTime = TimeGenerated, DeviceID = tostring(DeviceDetail.deviceId), DeviceName = tostring(DeviceDetail.displayName)
+    | project AADSignInTime, AppDisplayName, UserPrincipalName, UserDisplayName, Location, DeviceID, DeviceName, IPAddress
+    | distinct AADSignInTime, AppDisplayName, UserPrincipalName, UserDisplayName, Location, DeviceID, DeviceName, IPAddress //Remove duplicate events
+  ;
+  let ConnectAzAccountEvents = SigninLogs
+    | where AppDisplayName == "Microsoft Azure PowerShell" and ResultType == "0" and IsInteractive == true
+    | extend AzSignInTime = TimeGenerated, DeviceID = tostring(DeviceDetail.deviceId), DeviceName = tostring(DeviceDetail.displayName)
+    | project AzSignInTime, AppDisplayName, UserPrincipalName, UserDisplayName, Location, DeviceID, DeviceName, IPAddress
+    | distinct AzSignInTime, AppDisplayName, UserPrincipalName, UserDisplayName, Location, DeviceID, DeviceName, IPAddress //Remove duplicate events
+  ;
+  ConnectAzureADEvents
+  | join kind=leftouter (
+      ConnectAzAccountEvents
+      ) on UserPrincipalName
```

Comment:
Might want to consider joining on to include Location, DeviceID, DeviceName, IPAddress also, this would show you different devices completing the action under the same account.

Also, I would recommend either an additional project statement here to drop the Location1, DeviceID1, DeviceName1, IPAddress1 fields from the join.
OR
Rename the fields to something unique if you want to bring them thru.  For example - AzSigninLocation = Location1.

Reply:
Included Location, DeviceID and IPAddress